src/swig/nlopt.i

```diff
@@ -18,13 +19,18 @@ can be found at the NLopt web page: http://ab-initio.mit.edu/nlopt"
 %include "std_except.i"
 %include "std_vector.i"
 namespace std {
+#ifdef SWIGJAVA
+  %template(doublevector) vector<double>;
+#else
   %template(nlopt_doublevector) vector<double>;
+#endif
 };
 
 %ignore nlopt::opt::myfunc_data;
 %ignore nlopt::opt::operator=;
 
 // dont use the in-place version of get_initial_step
+%ignore nlopt_get_initial_step;
```

Comment:
why is this necessary ? isnt it a binding of the c++ interface, not the C one ?

Reply:
Because SWIG was generating a binding for that function (using ugly `SWIGTYPE_p_*` types) without this line. I am not sure why that particular function was getting picked up and none of the other C interface function, but I guess the definition somehow ends up transitively included by `nlopt.hpp`.

Reply:
I have checked where that `nlopt_get_initial_step` function comes from: That function is defined in `nlopt.hpp`:
```c++
// convenience overloading for below (not in nlopt:: since has nlopt_ prefix)
inline nlopt_result nlopt_get_initial_step(const nlopt_opt opt, double *dx) {
      return nlopt_get_initial_step(opt, (const double *) NULL, dx);
}
```
so it is part of the C++ interface. It cannot be implemented in the C interface because it overloads the C function with differing arguments.

I am not sure why this overload is defined at all. The comment implies that is a helper used in the C++ inline methods below, but it is not actually used there. The `opt::get_initial_step` method calls the regular 3-argument `nlopt_get_initial_step` from the C interface instead.

In any case, it is in-place and therefore should be excluded from all bindings.